test/arrayops.jl

```diff
@@ -2851,6 +2851,12 @@ end
     @inferred accumulate(*, String[])
     @test accumulate(*, ['a' 'b'; 'c' 'd'], dims=1) == ["a" "b"; "ac" "bd"]
     @test accumulate(*, ['a' 'b'; 'c' 'd'], dims=2) == ["a" "ab"; "c" "cd"]
+
+    # #53438
+    v = [(1, 2), (3, 4)]
+    @test_throws "does not return" accumulate(+, v)
```

Comment:
First commit includes the original error message for this case. I can squash that if folks prefer.